src/jit/morph.cpp

```diff
@@ -15322,11 +15321,16 @@ bool Compiler::fgMorphBlockStmt(BasicBlock* block, GenTreePtr stmt DEBUGARG(cons
 
     if (!removedStmt)
     {
-        /* Have to re-do the evaluation order since for example
-         * some later code does not expect constants as op1 */
+        if (lvaLocalVarRefCounted)
+        {
+            // fgMorphTree may have introduced new lclVar references. Bump the ref counts if requested.
+            lvaRecursiveIncRefCounts(stmt->gtStmtExpr);
```

Comment:
Can you also add some new logging here. to help us track down any potential fallout from this new change.
Like this:
#ifdef DEBUG
             printf("IExtra increment of LclVar refCounts for %s \n", msg);
#endif